Add InfoLogger type alias for v0.1.0 compatibility

v0.2.0 removed the InfoLogger interface in favor of returning a Logger from V(). This simplification is a nice improvement, but unfortunately this is a breaking change that makes it impossible to both satisfy the v0.1.0 and the v0.2.0 interface.

I am currently working on a Kubernetes operator using kubebuilder. When I switched to my own logr implementation ([genericr](https://github.com/wojas/genericr)), I found that Kubernetes depends on the v0.1.0 interface, while my backend implements the new v0.2.0 interface. I hit the following errors:

```
# sigs.k8s.io/controller-runtime/pkg/log
../../go/pkg/mod/sigs.k8s.io/controller-runtime@v0.6.0/pkg/log/null.go:48:32: undefined: logr.InfoLogger
# k8s.io/klog/v2
../../go/pkg/mod/k8s.io/klog/v2@v2.0.0/klog.go:690:45: undefined: logr.InfoLogger
../../go/pkg/mod/k8s.io/klog/v2@v2.0.0/klog.go:702:43: undefined: logr.InfoLogger
../../go/pkg/mod/k8s.io/klog/v2@v2.0.0/klog.go:706:48: undefined: logr.InfoLogger
../../go/pkg/mod/k8s.io/klog/v2@v2.0.0/klog.go:721:44: undefined: logr.InfoLogger
../../go/pkg/mod/k8s.io/klog/v2@v2.0.0/klog.go:739:55: undefined: logr.InfoLogger
../../go/pkg/mod/k8s.io/klog/v2@v2.0.0/klog.go:857:43: undefined: logr.InfoLogger
../../go/pkg/mod/k8s.io/klog/v2@v2.0.0/klog.go:1210:10: undefined: logr.InfoLogger
```

A quick look at klog suggests that that one can be fixed by declaring an internal InfoLogger interface without breaking anything, but I still need to verify this.

The custom null logger in the controller-runtime is a bigger problem. It is not possible to write an implementation of logr.Logger that works both with 0.1.0 and with 0.2.0. This is not a problem in a project where you control all the dependencies, but it is if you depend on third party code that may have their own implementation (e.g., for testing) of this interface.

Unfortunately this is a transition that is hard to manage for third party libraries, because the whole ecosystem would have to upgrade to v0.2.0 of the interface at the same time.

The problem can easily be solved by adding the following type alias to logr:

```go
// InfoLogger provides compatibility with code that relies on the v0.1.0 interface
// Deprecated: use Logger instead. This will be removed in a future release.
type InfoLogger = Logger
```

This ensures that v0.1.0 implementations continue to work with a new logr v0.2.1. Once v0.2.1+ becomes the norm in the ecosystem, the type alias can be dropped in a future release.

--- a/logr.go
+++ b/logr.go
@@ -176,3 +176,7 @@ type Logger interface {
 	// (see the package documentation for more information).
 	WithName(name string) Logger
 }
+
+// InfoLogger provides compatibility with code that relies on the v0.1.0 interface
+// Deprecated: use Logger instead. This will be removed in a future release.
+type InfoLogger = Logger

